test/gui/shared/steps/steps.py

```diff
@@ -265,14 +281,39 @@ def step(context, receiver, resource, permissions):
     collaboratorShouldBeListed(context, receiver, resource, permissions)
 
 
-def collaboratorShouldBeListed(context, receiver, resource, permissions):
+def collaboratorShouldBeListed(
+    context, receiver, resource, permissions, receiverCount=0
+):
     resource = getResourcePath(context, resource)
     socketConnect = syncstate.SocketConnect()
     socketConnect.sendCommand("SHARE:" + resource + "\n")
     permissionsList = permissions.split(',')
-    test.compare(
-        str(waitForObjectExists(names.scrollArea_sharedWith_QLabel).text), receiver
+
+    waitForObject(
+        {
+            "container": names.sharingDialogUG_scrollArea_QScrollArea,
+            "name": "sharedWith",
+            "type": "QLabel",
+            "visible": 1,
+        }
+    )
+
+    # findAllObjects: This functionfinds and returns a list of object references identified by the symbolic or real (multi-property) name objectName.
+    sharedWithObj = findAllObjects(
+        {
+            "container": names.sharingDialogUG_scrollArea_QScrollArea,
+            "name": "sharedWith",
+            "type": "QLabel",
+            "visible": 1,
+        }
     )
+
+    #     we use sharedWithObj list from above while verifying if users are listed or not.
+    #     For this we need an index value i.e receiverCount
+    #     For 1st user in the list the index will be 0 which is receiverCount default value
+    #     For 2nd user in the list the index will be 1 and so on
```

Comment:
not a big deal but could you please fromat them?
```suggestion
    # we use sharedWithObj list from above while verifying if users are listed or not.
    # For this we need an index value i.e receiverCount
    # For 1st user in the list the index will be 0 which is receiverCount default value
    # For 2nd user in the list the index will be 1 and so on
```

Reply:
@saw-jan This is also done by black